content/en/docs/tasks/administer-cluster/controller-manager-leader-migration.md

```diff
@@ -0,0 +1,115 @@
+
+This guide walks you through the manual process of upgrading the control plane from `kube-controller-manager` with built-in cloud provider to running both `kube-controller-manager` and `cloud-controller-manager`. If you use a tool to administrator the cluster, please refer to the documentation of the tool and the cloud provider for more details.
+
+## {{% heading "prerequisites" %}}
+
+The control plane nodes should run `kube-controller-manager` with Leader Election enabled through `--leader-elect=true`, and an in-tree cloud privider set with `--cloud-provider` flag.`cloud-controller-manager` should not yet be deployed.
+
+Your out-of-tree cloud provider must have built a `cloud-controller-manager` with Leader Migration implmentation. If the cloud provider imports `k8s.io/cloud-provider` and `k8s.io/controller-manager` of version v0.21.0 or later, Leader Migration will be avaliable.
+
+This guide assumes that kubelet of each control plane node starts `kube-controller-manager` and `cloud-controller-manager` as static pods defined by their manifests. If the components run in a different setting, please adjust the steps accordingly.
+
+For authorization, this guide assumes that the cluser uses RBAC. If another authorization mode grants permissions to `kube-controller-manager` and `cloud-controller-manager` components, please grant the needed access in a way that matches the mode.
+
+<!-- steps -->
+
+### Grant access to Migration Lease
+
+The default permissions of the controller manager allow only accesses to their main Lease. In order for the migration to work, accesses to another Lease are required.
+
+You can grant `kube-controller-manager` full access to the leases API by modifying the `system::leader-locking-kube-controller-manager` role. This task guide assumes that the name of the migration lease is `cloud-provider-extraction-migration`.
+
+`kubectl patch -n kube-system role 'system::leader-locking-kube-controller-manager' -p '{"rules": [ {"apiGroups":[ "coordination.k8s.io"], "resources": ["leases"], "resourceNames": ["cloud-provider-extraction-migration"], "verbs": ["create", "list", "get", "update"] } ]}' --type=merge`
+
+Do the same to the `system::leader-locking-cloud-controller-manager` role.
+
+`kubectl patch -n kube-system role 'system::leader-locking-cloud-controller-manager' -p '{"rules": [ {"apiGroups":[ "coordination.k8s.io"], "resources": ["leases"], "resourceNames": ["cloud-provider-extraction-migration"], "verbs": ["create", "list", "get", "update"] } ]}' --type=merge`
+
+### Initial Leader Migration configuration
+
+Leader Migration requires a configuration file representing the state of controller-to-manager assignment. The following example configuration shows the assignment at this moment where `route`, `service`, and `cloud-node-lifecycle` controllers are started within `kube-controller-manager`.
```

Comment:
Worth mentioning this is the default/normal set of controllers to migrate.

Reply:
Need to add `As of version 1.21` because we may change that in future release

Reply:
@jiahuif can you explain what you mean? Is this example something we'd need to update for future releases?

Reply:
This three controllers will eventually be removed from `kube-controller-manager` once legacy cloud providers are removed from k/k. Using legacy cloud providers is deprecated, which is exactly why we need to migrate them to cloud-controller-manager.

Reply:
If we change the set of legacy controllers that need migrating, I'd expect SIG Cloud Provider to update this example as part of documenting the change. The “version 1.21” bit is built into the documentation (we don't mention on each page that the documentation relates to the docs version you're viewing).

Does that sound right / make sense?

Reply:
You're right. I think `At this point` should be good enough.